src/s_net.c

```diff
@@ -62,7 +63,21 @@ int addrinfo_get_list(struct addrinfo **ailist, const char *hostname,
                      AI_PASSIVE;    /* listen to any addr if hostname is NULL */
     portstr[0] = '\0';
     sprintf(portstr, "%d", port);
-    return getaddrinfo(hostname, portstr, &hints, ailist);
+    result = getaddrinfo(hostname, portstr, &hints, ailist);
+        /* There's currently a bug in the BSD libc where getaddrinfo()
+         * will return EAI_BADFLAGS for the AI_ALL and AI_V4MAPPED flags.
+         * NOTE: this also seems to affect Android!
+         * In practice, this means we can't use dual stack sockets,
+         * so we fall back to IPv4 networking... */
+#if defined(__ANDROID__) || defined(__FreeBSD__ ) || defined(__NetBSD__) || defined(__OpenBSD__)
```

Comment:
why is this protected by ifdefs?
or put otherwise: what's the drawback on other OSs, to try again IPv4 only, if the dual-stack attempt failed (for whatever reasons, bug or not)

Reply:
I first had this without the #ifdef. My reasoning for the #ifdef was that we should only use the fallback on platforms which have the known BSD libc bug and on other platforms report the error as is. On the other hand, those other platforms probably won't ever show this error in the first place... So yes, I might get rid of the #ifdef again and only keep the comment.

BTW, what about printing a warning to stderr the first time 'EAI_BADFLAGS` occurs?